pkg/tarball/tarball.go

```diff
@@ -0,0 +1,61 @@
+package tarball
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"io"
+	"os"
+	"path"
+
+	"github.com/pkg/errors"
+)
+
+func DecodeTarball(reader io.Reader, baseDir string) error {
+	gzStream, err := gzip.NewReader(reader)
+	if err != nil {
+		return errors.Wrap(err, "couldn't uncompress reader")
+	}
+
+	tarchive := tar.NewReader(gzStream)
+	for {
+		header, err := tarchive.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return errors.Wrap(err, "couldn't opening tarball from gzip")
+		}
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.MkdirAll(path.Join(baseDir, header.Name), os.FileMode(header.Mode)); err != nil {
```

Comment:
oh, one thing i've run into is that sometimes the `header.Name` starts with `./` and sometimes it does not. It might be good to call `filepath.Clean` on `header.Name` when it is used